CONTRIBUTING.md

```diff
@@ -17,6 +20,103 @@ As a reminder, all contributors are expected to follow our
+
+## General Tips
+[general-tips]: #general-tips
+From experience, the following recommendations help to get a software
+contribution into RIOT master faster:
+
+- **Ask around for help!** Either offline or through one of our communication
+  channels (see above). The earlier you check your feature design with other
+  people, the less likely it is that it is denied during the review process.
+- **Verify your concept early!** If you work on your own until the code
+  *looks* good enough to show publicly, you might miss some design flaws others
+  might have spotted earlier.
+- **Keep it simple!** Try to use what is already there and don't change existing
+  APIs if not absolutely necessary.
+- **Keep it small!** A PR with >1000 lines of changes will very likely make
+  even the most active reviewer put your review on their long to-do list.
+- **Keep it modular!** Make extensions to a feature or new features for a
+  platform optionally to use.
+- **Provide tests!** They should be comprehensible and easy to be executed.
+  Alternatively comprehensive testing procedures should be provided with your
+  pull request.
+
+## Help Wanted
+[help-wanted]: #help-wanted
+In case you're not really sure where to start, we've created a list of suggestions.
+
+### Documentation
+If you've found yourself struggling to understand a particular aspect, chances
+are you're not the first and won't be the last. Writing down what you've learned
+is a great way to recap your new knowledge and share it with others. You can
+also start to learn about RIOT by combing through existing documentation and
+fixing errors and typos. Any help with improving the documentation is greatly
+appreciated and makes a big difference to the RIOT project.
+After you've finished writing, please publish your documentation in one of the
+following ways, depending on its type.
+
+#### General knowledge, HOWTOs
+Articles that focus on design aspects or how to use a particular module should
+be contributed to the [RIOT wiki](https://github.com/RIOT-OS/RIOT/wiki). After
+you've added your entry, please share it on the riot-dev mailing list so
+everyone is aware of its existence (and thank you).
+If you'd like to document a solution to minor annoyances or common pitfalls,
+please do not hesitate to extend the [Troubleshooting wiki
+page](https://github.com/RIOT-OS/RIOT/wiki/Troubleshooting). Again, please share
+your additions with the riot-dev mailing list.
+
+#### Code comments, HOWTOs for particular projects
+Documentation that relates directly to the code at hand like the HOWTO files
+that can be found in some of the directories in ``RIOT/examples/`` or comments
+in the code itself should be submitted through a [pull request][pull-requests].
+
+If you're not sure about the correct way to submit your writing, please ask on
+the mailing list or open an issue saying which documentation is missing. The
+other RIOTers will help you find the right format.
+
+### Issues
```

Comment:
Please note that I think that this subsection ("Issues") and the subsection that follows ("Contribute code") somewhat duplicate what is already described in "Feature Requests", "Bug Reports", and "Pull Requests", just in a more concise form